src/govuk/tools/_compatibility.scss

```diff
@@ -34,3 +36,14 @@
     @error "Non existent product '#{$product}'";
   }
 }
+
+/// Temporary public version of _govuk-compatibility to suppress local warnings
+///
+/// @access public
```

Comment:
This is the Sass doc that will appear in the Sass API reference, so I think we need to maintain the existing docs here but add the deprecation notice (so that the deprecation notice appears in the API reference).

So I'd expect this'd get flipped to something like:

```scss
/// Temporary private version of govuk-compatibility to avoid deprecation warnings
///
/// @access private

@mixin _govuk-compatibility($product) {
  // ...
}

/// Conditional Compatibility Mixin
///
/// Selectively output a block (available to the mixin as @content) if a given
/// $product is also identified as being used in the project.
///
/// This can then be used to include styles that are only needed to override
/// styles provided by those other products (e.g. where govuk_template has a
/// very specific link selector that otherwise affects buttons).
///
/// @example scss
///   // Override .my-class if GOV.UK Template is also being used
///   @include govuk-compatibility(govuk_template) {
///     .my-class {
///       color: inherit;
///     }
///   }
///
/// @param {String} $product - Name of product that we are 'defending' against.
/// @content Passed content is outputted only if Frontend is being used with
///   this product
/// @throw Errors if product name is not recognised
/// @access public
/// @deprecated Will be removed in v5.0 with the rest of the compatibility mode
/// suite of tools and settings

@mixin govuk-compatibility($product) {
  // ...
}
```